### ISSUE TITLE
catch "if" statement with single "="

### ISSUE THREAD

**Issue by nickperkinslondon** [clutchski/coffeelint#355]
How about catching the mistake that I make most often?
An "if" statement that uses a single "=" instead of double "==" compiles just fine, but causes confusing behaviour at runtime.
I wish my editor would warn me when I do this.
Example:

```
if a = b
    do_something()
```

**Comment by nickperkinslondon** [comment]
I still think this should be a lint.
The "parens" solution would be ok, but I think you could just flag it always.
If the programmer actually intends to do assignment, it could easily be done before the "if":

    foo = bar()
    if foo
        baz(foo)

I realize that this will cause some "false positives", but it's all about catching the silly mistakes that humans make, and I think it is probably a common mistake.

( As a side-note, I have tried to train myself to use "is" instead of "==", which also helps avoid this problem, and is perhaps a more idiomatic way to write cs )
